pkg/config/config.go

```diff
@@ -222,6 +241,29 @@ func (cfg *Config) Validate() error {
 	if cfg.EnableWebhookServer && cfg.WebhookServerAddr == "" {
 		return errors.New("empty webhook server address")
 	}
+
+	if cfg.DefaultDriftRemediationPeriod < 0 {
+		return fmt.Errorf("invalid value for flag '%s': drift remediation seconds default must be greater than or equal to 0", flagDefaultDriftRemediationPerdiod)
+	}
+
+	if cfg.DriftRemediationConfigurationJSON != "" {
+		// Validate that the drift remediation configuration is a valid JSON string
+		var drc map[string]int
+		if err := json.Unmarshal([]byte(cfg.DriftRemediationConfigurationJSON), &drc); err != nil {
+			return fmt.Errorf("invalid value for flag '%s': %v", flagDriftRemediationConfigurationJSON, err)
+		}
+
+		// Validate that all resource-specific periods are greater than or equal to 0
```

Comment:
Can we confirm here whether the `kind` refers to a valid resource?

Reply:
Good suggestion! Unfortunately we cannot verify this at the configuration level. I can try to add a panic during controller startup (if a user provides a configuration for a non existing resource) WDYT ?

Reply:
Having second thoughts here, adding this might also require few changes in code-generator.. Reconcilers are only aware of one resources (the one they manage)

Reply:
My concern is that users might mis-spell the resource name on the command line or in the config and as things stand right now, there would be simply a silent failure to enforce the intended resync period.  Is there another way to mitigate that possible problem?  If not, then let's be sure to emphatically document that there isn't any error checking so users should be careful to get the names right.

Reply:
related: https://github.com/aws-controllers-k8s/runtime/pull/106#discussion_r1086940308

Reply:
https://github.com/aws-controllers-k8s/runtime/pull/106#discussion_r1088307180